packages/fxa-settings/src/components/ThirdPartyAuth/index.tsx

```diff
@@ -242,23 +247,33 @@ function deleteParams(searchParams: URLSearchParams, paramsToDelete: string[]) {
   return searchParams;
 }
 
-function getState() {
+function getState(flowQueryParams: QueryParams | undefined) {
   // We stash the originating location in the state oauth param
   // because we will need it to use it to reconstruct the redirect URL for RP
   const params = new URLSearchParams(window.location.search);
-  // we won't need these params that are used for internal backbone/react navigation
-  const modifiedParams = deleteParams(params, [
+
+  // Combine flowQueryParams and paramsObject, ensuring all values are strings
+  const paramsObject = Object.fromEntries(params.entries());
```

Comment:
Interesting.  Is there something special to copying an object this way?  (Instead of, say, `const clone = {...og}`.)